Kudu.Core/Deployment/StructuredTextLogger.cs

```diff
@@ -0,0 +1,131 @@
+﻿using Kudu.Core.Tracing;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Kudu.Core.Deployment
+{
+    public class StructuredTextLogger : IDetailedLogger
+    {
+        private readonly static object DocumentLock = new object();
+        private readonly IAnalytics _analytics;
+        private int _depth;
+        private readonly string _path;
+        private const string LogEntryTemplate = "message=\"{0}\" time=\"{1}\" id=\"{2}\" type=\"{3}\"";
+        private readonly StructuredTextDocument<LogEntry> _structuredTextDocument;
+
+        private static IEnumerable<KeyValuePair<string, string>> EscapeChars = new Dictionary<string, string>
+        {
+            { "\"", "&quot;" }
+        }
+        .Union(StructuredTextDocument.NotAllowedSequances);
+
+        public StructuredTextLogger(string path, IAnalytics analytics)
+        {
+            _depth = 0;
+            _path = path;
+            _analytics = analytics;
+            _structuredTextDocument = new StructuredTextDocument<LogEntry>(path,
+                //DateTime.ToString("o") => "2015-08-04T00:08:38.5489308Z"
+                e => string.Format(LogEntryTemplate, e.Message, e.LogTime.ToString("o"), e.Id, (int)e.Type),
+                str =>
+                {
+                    var splitted = str.Split(new[] { '"' }, StringSplitOptions.None);
+                    if (splitted.Length == 9)
+                    {
+                        var message = UnsanitizeValue(splitted[1]);
+                        var time = DateTime.Parse(splitted[3]).ToUniversalTime();
+                        var id = splitted[5];
+                        var type = (LogEntryType)Int32.Parse(splitted[7]);
+                        return new LogEntry(time, id, message, type);
+                    }
+                    else
+                    {
+                        throw new FormatException(string.Format("the log line \"{0}\" is in an invalid format", str));
+                    }
+                });
+        }
+
+        public ILogger Log(string value, LogEntryType type)
+        {
+            try
+            {
+                value = SanitizeValue(value);
+                lock(DocumentLock)
+                {
+                    _structuredTextDocument.Write(new LogEntry(DateTime.UtcNow, Guid.NewGuid().ToString(), value, type), _depth);
```

Comment:
Actually, I'm not sure that the ID provides any value at all, since we don't have any routes that gets an individual item. Though we need to be careful not to break existing clients (like portal) in case they look for this ID in the API.